### ISSUE TITLE
Add synced batchnorm support

### ISSUE THREAD

**Issue by williamFalcon** [PyTorchLightning/pytorch-lightning#2589]
https://pytorch.org/docs/stable/nn.html#torch.nn.SyncBatchNorm

![image](https://user-images.githubusercontent.com/3640001/87235279-6be03800-c3a8-11ea-8fd9-efdcebed372b.png)

@justusschock or @awaelchli interested?

**Comment by ananthsub** [comment]
Should this also support apex's `create_syncbn_process_group`: https://nvidia.github.io/apex/_modules/apex/parallel.html

**Comment by ananyahjha93** [comment]
@williamFalcon adding this
